File: 200-number-of-islands/200-number-of-islands.java
class Solution {
    
    private void helper(char[][] grid, int x, int y, int m, int n){
        if(x<0||x>=m||y<0||y>=n||grid[x][y]=='0'||grid[x][y]=='2'){
            return;
        }
        grid[x][y] = '2';
        helper(grid,x+1,y,m,n);
        helper(grid,x,y+1,m,n);
        helper(grid,x-1,y,m,n);
        helper(grid,x,y-1,m,n);
    }
    public int numIslands(char[][] grid) {
        int m = grid.length;
        int n = grid[0].length;
        int count = 0;
        for(int i=0;i<m;i++){
            for(int j=0;j<n;j++){
                if(grid[i][j]=='1'){
                    //System.out.println("hehe");
                    helper(grid,i,j,m,n);
                    count++;
                }
            }
        }
        return count;
    }
}

File: 200-number-of-islands/200-number-of-islands.py
class Solution:
    def numIslands(self, grid: List[List[str]]) -> int:
        
        def recurse(grid,i,j,n,m):
            if i<0 or i==n or j<0 or j==m or grid[i][j]=="-1" or grid[i][j]=="0":
                return
            grid[i][j] = "-1"
            recurse(grid,i+1,j,n,m)
            recurse(grid,i,j+1,n,m)
            recurse(grid,i-1,j,n,m)
            recurse(grid,i,j-1,n,m)
        ans = 0
        for i in range(len(grid)):
            for j in range(len(grid[0])):
                if grid[i][j]=="1":
                    recurse(grid,i,j,len(grid),len(grid[0]))
                    ans += 1
        return ans
                    